src/Contexts/PrintLoadingContext.tsx

```diff
@@ -33,50 +33,64 @@ export enum PrintLoadingActions {
     PRINT,
 }
 
-const reducer = (state: PrintLoadingState, action: PrintLoadingDispatch): PrintLoadingState => {
-    switch(action.type) {
-    case PrintLoadingActions.RESET_EXPECTED_COUNT:
-        logger.debug('Printing state reset.');
-        return {expected: 1, arr: []};
-    case PrintLoadingActions.ADD_EXPECTED_PROMISE_COUNT:
-        return {expected: state.expected + action.expected, arr: state.arr};
-    case PrintLoadingActions.ADD_PROMISE_INCR_EXPECTED:
-        logger.debug('Promise received');
-        // You must use the other action to increase expected counts.
-        return {expected: state.expected+1, arr: [...state.arr, action.payload]};
-    case PrintLoadingActions.ADD_PROMISE:
-        logger.debug('Promise received');
-        // You must use the other action to increase expected counts.
-        return {expected: state.expected, arr: [...state.arr, action.payload]};
-    case PrintLoadingActions.PRINT:
-        if (state.expected !== action.expected) {
-            logger.debug(`Skipping early print call from ${action.expected} -> ${state.expected}`);
-            return state;
-        }
-        logger.debug(`Printing because ${state.expected} == ${action.expected}`);
-        print();
-        return state;
-    }
-    return state;
-};
-
 export const usePrintLoadingContext = () => React.useContext(PrintLoadingContext);
 
 export const PrintLoadingProvider: React.FC<Props>  = ({ children }) => {
-    const [state, dispatch] = useReducer(reducer, {expected: 1, arr: []});
+    const timeout = React.useRef<NodeJS.Timeout | null>(null);
+    const state: PrintLoadingState = {expected: 0, arr: []};
```

Comment:
Should this be a use ref?
Also we should probably rename from state since it is confusing